webdriver/tests/bidi/network/conftest.py

```diff
@@ -44,7 +44,7 @@ def fetch(bidi_session, top_context, configuration):
     """Perform a fetch from the page of the provided context, default to the
     top context.
     """
-    async def fetch(url, method="GET", headers=None, context=top_context, timeout_in_seconds=3):
+    async def fetch(url, method="GET", headers=None, context=top_context, timeout_in_seconds=0.5):
```

Comment:
So this is doing the actual fetch in the browser? So if we have e.g. a debug or tsan build of the browser it might be slow? But we are applying the timeout multiplier, so maybe that's accounted for.

In general I don't see why the fetch itself would time out. Do we have examples of this causing a problem?

Reply:
So how most test that use the following:
1. Set-up listeners for network event(s).
2. Run this `fetch` function and wait for Promise to complete/fail or timeout (when no response is received)
3. Check events data.

And example would be what is happening in Chrome for `network.responseCompleted` test `test_response_status`.
There there seem to be some quirkiness with Network request with `101` and `407` statuses as they never seem to resolve.

Reply:
Please do not hard-code. Instead we should make it similar to what we have in the [`session` fixture for WebDriver classic](https://searchfox.org/mozilla-central/source/testing/web-platform/tests/webdriver/tests/support/fixtures.py#180-184) and (as James said) use the multiplier at least. I'm not sure of a good default value but I feel that both the fetch and the navigate timeout should have potentially the same value by default?

Reply:
> In general I don't see why the fetch itself would time out. Do we have examples of this causing a problem?

There are some tests in network interception where we explicitly rely on the timeout to check that the request was blocked. We don't do it for all the tests, but I thought it was nice to test that there is an actual request timeout. That is done on top of checking that the events have the appropriate isBlocked flag and that no subsequent events are received.

For those specific tests, we could definitely reduce the timeout when calling fetch.